.travis.yml

```diff
@@ -15,54 +15,57 @@ before_install:
- - scons compileMode=release order=6 parallelization=hybrid commThread=yes arch=dsnb compiler=gcc unitTests=fast check
-   # netcdf is built statically with hdf5 support, hence the following do not link
-   # - scons compileMode=release order=6 parallelization=hybrid netcdf=yes arch=dsnb compiler=gcc unitTests=fast -j2
-   # - scons compileMode=release order=6 parallelization=hybrid netcdf=yes arch=dsnb compiler=gcc unitTests=fast check
- - scons compileMode=release order=6 parallelization=hybrid hdf5=yes arch=dsnb compiler=gcc unitTests=fast -j2
- - scons compileMode=release order=6 parallelization=hybrid hdf5=yes arch=dsnb compiler=gcc unitTests=fast check
- - scons compileMode=release order=6 parallelization=hybrid netcdf=yes hdf5=yes metis=yes commThread=yes arch=dsnb compiler=gcc unitTests=fast -j2
- - scons compileMode=release order=6 parallelization=hybrid netcdf=yes hdf5=yes metis=yes commThread=yes arch=dsnb compiler=gcc unitTests=fast check
- - scons compileMode=release order=6 parallelization=none arch=dsnb compiler=gcc unitTests=fast -j2
- - scons compileMode=release order=6 parallelization=none arch=dsnb compiler=gcc unitTests=fast check
- - scons compileMode=debug order=6 parallelization=hybrid arch=dsnb compiler=gcc unitTests=fast -j2
- - scons compileMode=debug order=6 parallelization=hybrid arch=dsnb compiler=gcc unitTests=fast check
- - scons compileMode=release order=6 parallelization=hybrid arch=ssnb compiler=gcc unitTests=fast -j2
- - scons compileMode=release order=6 parallelization=hybrid arch=ssnb compiler=gcc unitTests=fast check
- - scons compileMode=release order=6 parallelization=hybrid netcdf=yes hdf5=yes commThread=yes arch=ssnb compiler=gcc unitTests=fast -j2
- - scons compileMode=release order=6 parallelization=hybrid netcdf=yes hdf5=yes commThread=yes arch=ssnb compiler=gcc unitTests=fast check
- - scons compileMode=debug order=6 parallelization=hybrid arch=ssnb compiler=gcc unitTests=fast -j2
- - scons compileMode=debug order=6 parallelization=hybrid arch=ssnb compiler=gcc unitTests=fast check
+ - |
+   set -euo pipefail
+   export IFS=$'\n\t'
+   export CTEST_OUTPUT_ON_FAILURE=1
+   for equation in elastic viscoelastic2 anisotropic; do
+    if [ "$equation" = viscoelastic2 ]; then
+        mechanisms=3
+    else
+        mechanisms=0
+    fi
+        dirname="build_${equation}_${SEISSOL_PRECISION}_${SEISSOL_BUILD_TYPE}"
+        echo "mkdir ${dirname}"
+        mkdir -p $dirname && cd $dirname
+        CMAKE_PREFIX_PATH=~ PKG_CONFIG_PATH=~/lib/pkgconfig/: cmake -DNETCDF=OFF -DMETIS=ON -DCOMMTHREAD=ON -DASAGI=OFF -DHDF5=ON -DCMAKE_BUILD_TYPE=$SEISSOL_BUILD_TYPE -DTESTING=ON -DLOG_LEVEL=warning -DLOG_LEVEL_MASTER=info -DHOST_ARCH=hsw -DPRECISION=$SEISSOL_PRECISION -DEQUATIONS=$equation -DNUMBER_OF_MECHANISMS=$mechanisms -DGEMM_TOOLS_LIST=LIBXSMM ..
+        make -j $(nproc)
+        make test
+        cd ..
+    done
+    set +u
```

Comment:
Is there a reason why you have to set +u at the end? I'd assume that it's not doing anything here?

Reply:
It is actually needed. *u* in the following line _set -euo pipefail_ forces bash to treat unset variables and parameters other than the special parameters ‘@’ or ‘*’ as an error when performing parameter expansion.There are some unused parameters in bash functions used by Travis. Therefore, I had to restore the settings back. This Travis bug has been reported and mentioned in some threads dedicated to Travis CI. Currently, it is the solution.

Reply:
I see. I knew what +euo pipefail does - I never write a bash script without, as not setting it can hide errors. Didn't know that that breaks travis. Thanks for figuring this out!